### PR TITLE
Do not derive Copy for EpochSchedule and Rent

### LINKED ISSUE #32588: CloneZero macro and Copy types
#### Problem
See: https://github.com/solana-labs/solana/pull/32547#discussion_r1271091160

In summary a new clippy lint does not like when `Clone` is not simply `Copy` for `Copy`-types.
We added a clippy allow to pass clippy checks using future versions of cargo.

But we discussed evaluating whether these types should even be `Copy`. 

#### Proposed Solution
Try removing `Copy` on `CloneZeroed` types

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
In #32547, I added a clippy allow lint for `CloneZeroed` types that were `Copy` to have an "incorrect" clone. For any `Copy` types, clippy considers a `clone` that is anything other than `Copy` to be incorrect.

`EpochSchedule` and `Rent` turned out to be the only `Copy` types using this derive macro.

#### Summary of Changes
- EpochSchedule no longer derives `Copy`
- Rent no longer derives `Copy`
- Replace use of `Copy` by `Clone` (this will ensure padding bytes are zero consistently)
- Use references where appropriate	

Fixes #32588

### CODE REVIEW

File: ledger/src/leader_schedule_cache.rs
Comment (apfitzge) [reply ledger/src/leader_schedule_cache.rs]: done: 79ca5b60a0

**Comment by apfitzge** [comment]
this got stale because there was a merge conflict I didn't resolve for a bit.

I think some of this may conflict with changes #32961 (or related PRs) has. So going to just hold this off until those get merged.

**Comment by Lichtso** [comment]
@apfitzge I think you can rebase

**Comment by apfitzge** [comment]
Some downstream issues w/ removing `Copy` from EpochSchedule in stake_pool. Will have to look at how that's being used

**Comment by apfitzge** [comment]
SPL fix PR: https://github.com/solana-labs/solana-program-library/pull/5323

**Comment by apfitzge** [comment]
> SPL fix PR: [solana-labs/solana-program-library#5323](https://github.com/solana-labs/solana-program-library/pull/5323)

Hmm must have been another issue in SPL. I had run `cargo test-sbf` which I thought would have run all the tests. Will have to check again tomorrow...

**Comment by apfitzge** [comment]
Finally fixed all the SPL stuff 😦 @Lichtso, @brooksprumo

**Comment by apfitzge** [comment]
@Lichtso had to rebase this again, but should be in a good state again now.

**Review by Lichtso — APPROVED** [review]

**Pull request merged**
